src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -774,8 +759,62 @@ void Engine::ApplyPerformanceHints(std::map<std::string, std::string> &config, c
             num_streams = std::min(num_streams,
                                    engConfig.perfHintsConfig.ovPerfHintNumRequests);
         }
-        config[CONFIG_KEY(CPU_THROUGHPUT_STREAMS)] = std::to_string(num_streams);
-        config[ov::num_streams.name()] = ov::util::to_string(num_streams);
+        return std::pair<std::string, std::string>(std::to_string(num_streams), ov::util::to_string(num_streams));
+    };
+
+    auto getModeName = [&]() {
+        const bool streamsExplicitlySetForModel = streamsSet(config);
+        // checking streams (to avoid overriding what user might explicitly set in the incoming config or previously via SetConfig)
+        if (streamsExplicitlySetForModel ||
+            streamsExplicitlySetForEngine)
+            return std::string();
+
+        const auto& mode = config.find(CONFIG_KEY(PERFORMANCE_HINT));
+        // the mode may have just arrived to the LoadNetwork, or was set with the plugin's SetConfig
+        if (mode == config.end() && engConfig.perfHintsConfig.ovPerfHint.empty())
+            return std::string();
+        /* performance hints set for network has higher pririty than engine ones.
+        * This applies for all the configuration parameters */
+        const auto mode_name = (mode != config.end()) ?
+            PerfHintsConfig::CheckPerformanceHintValue(mode->second) :
+            engConfig.perfHintsConfig.ovPerfHint;
+        return mode_name;
+    };
+
+    const auto& modelCacheProp = config.find(InferenceEngine::PluginConfigParams::KEY_CACHE_DIR);
+    const bool enableModelCache = (modelCacheProp != config.end() && !modelCacheProp->second.empty()) || !engConfig.cache_dir.empty();
+    // We compute both hints values if caching enabled to avoid different cache files for different hints mode
+    if (enableModelCache) {
```

Comment:
This condition doesn't cover pure Import/Export path (w/o caching). I don't see test cases as well.

Reply:
Corrected, thanks